GoogleForCreators/web-stories-wp
# Editor: Add Semantic Headings to the right click menu of Text Elements

## Context

Within the right click menu of a text element on the page, there should be a new item called Heading Level
This Heading Level item should have a submenu that shows the following properties (along with the one currently selected):
- `Automatic (xyz)` (stored as auto)
- `Heading 1` (stored as `h1`)
- `Heading 2` (stored as `h2`)
- `Heading 3` (stored as `h3`)
- `Paragraph` (stored as `p`)



## Summary

Adds right click menu for heading level to text elements.

<img width="517" alt="Screen Shot 2022-08-03 at 5 39 10 AM" src="https://user-images.githubusercontent.com/62242/182577314-ea6bf3c0-92b8-4128-92be-99f32de57a9c.png">





## Relevant Technical Choices



## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Add a text element to the canvas
2. Right click the element
3. Select Heading Level from the `Right Click Menu`.
4. Select a new heading level from the sub menu and note that it gets selected when click
5. Verify the style panel heading level matches what is selected from the `Right Click Menu`.
6. Set the heading level to `Automatic` - note that the "current" tag name is noted for the "Automatic" option

## Reviews

### Does this PR have a security-related impact?



### Does this PR change what data or activity we track or use?



### Does this PR have a legal-related impact?



## Checklist




---



Fixes #11402
